tests/test_devices.py

```diff
@@ -6,86 +6,87 @@
     )
 str = type('')
 
+import warnings
 
 import pytest
 
-from gpiozero.pins.mock import MockPin
 from gpiozero import *
 
 
 def teardown_function(function):
-    MockPin.clear_pins()
+    Device._pin_factory.reset()
+
 
 # TODO add more devices tests!
 
-def test_device_no_pin():
+def test_device_bad_pin():
     with pytest.raises(GPIOPinMissing):
         device = GPIODevice()
+    with pytest.raises(PinInvalidPin):
+        device = GPIODevice(60)
+
+def test_device_non_physical():
+    with warnings.catch_warnings(record=True) as w:
+        device = GPIODevice(37)
+        assert len(w) == 1
+        assert w[0].category == PinNonPhysical
 
 def test_device_init():
-    pin = MockPin(2)
+    pin = Device._pin_factory.pin(2)
     with GPIODevice(pin) as device:
         assert not device.closed
         assert device.pin == pin
 
 def test_device_init_twice_same_pin():
-    pin = MockPin(2)
-    with GPIODevice(pin) as device:
+    with GPIODevice(2) as device:
         with pytest.raises(GPIOPinInUse):
-            device2 = GPIODevice(pin)
+            GPIODevice(2)
 
 def test_device_init_twice_different_pin():
-    pin = MockPin(2)
-    pin2 = MockPin(3)
-    with GPIODevice(pin) as device:
-        with GPIODevice(pin2) as device2:
+    with GPIODevice(2) as device:
+        with GPIODevice(3) as device2:
             pass
 
 def test_device_close():
-    pin = MockPin(2)
-    device = GPIODevice(pin)
+    device = GPIODevice(2)
```

Comment:
Looking at the other tests, should tests like this also be refactored to `with GPIODevice(2) as device:`?
Is there any appreciable difference, or is it just a stylistic thing?